This patch fixes several issues with WebGL on NVIDIA/X11:

    * Fix eglGetCurrentContext issue in ANGLE with NVIDIA driver
      See https://bugs.webkit.org/show_bug.cgi?id=261653

    * Fix segmentation fault with libepoxy;

    * Set m_anglePlatform for X11.

--- webkitgtk-2.42.0/Source/ThirdParty/ANGLE/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp~	2023-08-09 09:49:33.615939600 +0000
+++ webkitgtk-2.42.0/Source/ThirdParty/ANGLE/src/libANGLE/renderer/gl/egl/DisplayEGL.cpp	2023-09-30 12:58:47.043872685 +0000
@@ -945,7 +945,7 @@
         ASSERT(!makeNewContextCurrent);
         // TODO(penghuang): Should we consider creating a share context to avoid querying and
         // restoring GL context state? http://anglebug.com/5509
-        context = mEGL->getCurrentContext();
+        context = mCurrentNativeContexts[angle::GetCurrentThreadUniqueId()].context;
         ASSERT(context != EGL_NO_CONTEXT);
         // TODO(penghuang): get the version from the current context. http://anglebug.com/5509
         attribs = {EGL_CONTEXT_MAJOR_VERSION, 2, EGL_CONTEXT_MINOR_VERSION, 0, EGL_NONE};
--- webkitgtk-2.42.0/Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLENicosia.cpp~	2023-09-26 10:51:52.915360944 +0000
+++ webkitgtk-2.42.0/Source/WebCore/platform/graphics/texmap/GraphicsContextGLTextureMapperANGLENicosia.cpp	2023-09-26 10:52:08.999307697 +0000
@@ -27,7 +27,7 @@
 #include "GraphicsContextGLTextureMapperANGLE.h"
 
 #if ENABLE(WEBGL) && USE(TEXTURE_MAPPER) && USE(NICOSIA)
-#include <epoxy/gl.h>
+#include <GL/gl.h>
 
 namespace WebCore {
 
--- webkitgtk-2.42.0/Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp-	2023-09-26 12:55:29.301613100 +0000
+++ webkitgtk-2.42.0/Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp	2023-09-26 12:53:59.971122612 +0000
@@ -69,6 +69,9 @@
 PlatformDisplayX11::PlatformDisplayX11(Display* display)
     : m_display(display)
 {
+#if ENABLE(WEBGL)
+    m_anglePlatform = EGL_PLATFORM_X11_KHR;
+#endif
 }
 
 #if PLATFORM(GTK)
@@ -76,6 +79,9 @@
     : PlatformDisplay(display)
     , m_display(display ? GDK_DISPLAY_XDISPLAY(display) : nullptr)
 {
+#if ENABLE(WEBGL)
+    m_anglePlatform = EGL_PLATFORM_X11_KHR;
+#endif
 }
 #endif
 
